core/trino-main/src/main/java/io/trino/server/security/CertificateAuthenticator.java

```diff
@@ -55,6 +56,14 @@ public Identity authenticate(ContainerRequestContext request)
         }
 
         try {
+            Optional<Identity> identityOpt = authenticatorManager.getAuthenticator().createAuthenticatedIdentity(certificates);
```

Comment:
I believe we need to validate if the ideniity is correct.  For example don't we expect that principal is set?